spec/jobs/idv/profile_job_spec.rb

```diff
@@ -45,6 +75,28 @@
         expect(result.reasons).to eq(['The name was suspicious'])
         expect(result.errors).to eq(first_name: 'Unverified first name.')
         expect(result.session_id).to be_present
+        expect(agent).to have_received(:start)
+        expect(agent).to_not have_received(:submit_state_id)
+      end
+    end
+
+    context 'when state id confirmation fails' do
+      let(:vendor_params) { super().merge(state_id_number: '000000000') }
+
+      it 'saves an unsuccessful result' do
+        Idv::ProfileJob.perform_now(
+          result_id: result_id,
+          vendor_params: vendor_params,
+          applicant_json: applicant_json
+        )
+        result = VendorValidatorResultStorage.new.load(result_id)
+
+        expect(result.success?).to eq(false)
+        expect(result.timed_out?).to eq(false)
+        expect(result.job_failed?).to eq(false)
+        expect(result.reasons).to eq(['Everything looks good', 'invalid state id number'])
```

Comment:
I'm not sure if this would be out of scope for this PR,  but would it be possible to either: 
1) Add the type to the reasons (for example, "resolution" and "state_id_confirmation") 
or 
2) only include failure reasons when the result is not successful?

It can be confusing to see `Everything looks good` and `invalid state id number` in the same array.